Models/RegistrationBase.php

```diff
@@ -359,12 +363,108 @@ public static function create($request)
         $reg->fill($inputs);
         $reg->save();
 
+
         $response['success'] = true;
         $response['data']['item'] = $reg;
         $response['messages'][] = trans('vaahcms-general.saved_successfully');
         return $response;
 
     }
+
+    //-------------------------------------------------
+
+    public static function createRegistration($request)
+    {
+
+        $inputs = $request->all();
+
+           $rules = [
+                'first_name' => 'required|max:150',
+                'last_name' => 'required|max:150',
+                'username' => 'required|max:150',
+               'email' => 'required|regex:/(.+)@(.+)\.(.+)/i',
+                'password' => [
+                    'required',
+                ],
+                'confirm_password' => 'required|same:password',
+            ];
+
+            $messages = [
+                'confirm_password.same' => 'Password and confirm password does not match.',
+            ];
+
+            $validator = \Validator::make($request->all(), $rules, $messages);
+
+        if ( $validator->fails() ) {
+            $errors = errorsToArray($validator->errors());
+            $response['success'] = false;
+            $response['errors'] = $errors;
+            return $response;
+        }
+
+        // check if already exist
+        $user = static::where('email',$inputs['email'])->first();
+
+        if($user)
+        {
+            $response['success'] = false;
+            $response['messages'][] = trans('vaahcms-user.email_already_registered');
+            return $response;
+        }
+
+        // check if username already exist
+        $user = self::where('username',$inputs['username'])->first();
+
+        if($user)
+        {
+            $response['success'] = false;
+            $response['messages'][] = trans('vaahcms-user.username_already_registered');
+            return $response;
+        }
+
+        // check if user already exist
+        $user = User::where('email',$inputs['email'])->first();
```

Comment:
@we-anil-y001 you've already written is query above, why are you writing it again?

Reply:
sir it check from vh_users table email exist or not and above check form vh_registration table.